### PR TITLE
Add logs to cleaning steps in the ingestion server and skip saving tags

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Follow up on #4163.

## Description

Add several logging points to determine what happens with this step that hangs with production data. Also, skip saving tags to see if we can get the first files for URL fields.

## Testing Instructions

1. make some rows of the image table in the catalog dirty by removing the protocol in one or several of the URLs (url, creator_url, or foreign_landing_url) or modify the tags for low accuracy.
2. `just catalog/up && just api/init`
3. Observe the logs of the ingestion server container.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (`just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by AetherUnbound — COMMENTED** [review]
LGTM and worked locally, just one note about a conditional that appears out of place (or, at least, is not listed in the PR description as an expected set of changes).

File: ingestion_server/ingestion_server/cleanup.py [hunk 1]
```diff
@@ -255,6 +255,10 @@ def _clean_data_worker(rows, temp_table, sources_config, all_fields: list[str]):
         for field, clean_value in cleaned_data.items():
             update_field_expressions.append(f"{field} = {clean_value}")
             # Save cleaned values for later
+            # (except for tags, which take up too much space)
+            if field == "tags":
+                log.debug("Skipping tags.")
+                continue
```
[the remark this answers is not in the document]
Comment (krysal) [reply ingestion_server/ingestion_server/cleanup.py:261]: Yes, I forgot to mentioning it. I want to test if the hanging has something to do with tags, so I'll omit it _for now_.

**Review by obulat — APPROVED** [review]
LGTM!

File: ingestion_server/ingestion_server/cleanup.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply ingestion_server/ingestion_server/cleanup.py:261]: Okay! Do you mind adding that to the PR title and maybe the description to make sure that change is captured in the final commit?

[the remark this answers is not in the document]
Comment (krysal) [reply ingestion_server/ingestion_server/cleanup.py:261]: Sure, done!

**Review by AetherUnbound — APPROVED** [review]

**Pull request merged**